src/libs/conduit/conduit_node.cpp

```diff
@@ -13501,6 +13502,146 @@ Node::find_first_data_ptr() const
+
+            Node &info_child = info[child_path];
+            if(!n.has_child(child_path))
+            {
+                info_child.set("Missing from other.");
+                res = false;
+            }
+            else
+            {
+                if(m_child.compare(n.fetch(child_path), info_child))
+                {
+                    info.remove(child_path);
+                }
+                else
+                {
+                    res = false;
+                }
+            }
+        }
+
+        child_itr = n.children();
+        while(child_itr.has_next())
+        {
+            const conduit::Node &n_child = child_itr.next();
+            const std::string child_path = child_itr.name();
+
+            Node &info_child = info[child_path];
+            if(!has_child(child_path))
+            {
+                info_child.set("Missing from self.");
+                res = false;
+            }
+            else
+            {
+                if(fetch(child_path).compare(n_child, info_child))
+                {
+                    info.remove(child_path);
+                }
+                else
+                {
+                    res = false;
+                }
+            }
+        }
+    }
+    else if(m_dtid == DataType::LIST_ID)
+    {
+        // NOTE: This case is a bit strange in the comparison; fortunately, it
+        // can be distilled to two primary cases:
+        // - Nodes Have 0 Diffs: The info node is completely empty.
+        // - Nodes Have >1 Diffs: The info node is a list where the differing
+        //   nodes between the comparison parents are filled in the list.
+        //   All other nodes in the list are empty.
+
+        index_t m_nchild = number_of_children();
+        index_t n_nchild = n.number_of_children();
+
+        size_t i = 0;
+        for(; i < (size_t)std::min(m_nchild, n_nchild); i++)
+        {
+            const Node &m_child = child(i);
+            const Node &n_child = n.child(i);
+            Node &info_child = info.append();
+            res &= m_child.compare(n_child, info_child);
+        }
+
+        res &= m_nchild != n_nchild;
+        for(; i < (size_t)std::max(m_nchild, n_nchild); i++)
+        {
+            Node &info_child = info.append();
+            std::string loc_str = (i >= (size_t)m_nchild) ? "self" : "other";
+            info_child.set("Missing from " + loc_str);
+        }
+    }
+    else if(!dtype().compatible(n.dtype()) || !n.dtype().compatible(dtype()))
+    {
+        info.set("Data type incompatibility.");
+        res = false;
+    }
+    else
+    {
+        // TODO(JRC): Determine how to customize this function so that users that
+        // want the stride/endianness to be the same can still perform a proper query.
+        DataType elem_dtype(dtype().id(), 1, 0, 0, dtype().element_bytes(), dtype().endianness());
+        for(size_t i = 0; i < (size_t)dtype().number_of_elements(); i++)
```

Comment:
element_ptr takes care of the stride
 
I don't think we need to worry about endianness. b/c if endianness is different any actual value comparison will fail (there isn't magical endianness conversion when accessing values, users must do it explicitly).

Some thoughts:

The most useful outcome would be a list of what is different between the two arrays, we may want a bytes vs numeric option. 


Can we just diff the bytes and then for output drop into special cases for numeric differences (dtype().is_number()?).

Should we store the all differences, or store the differences compactly (element 2 differs by -.003343, element 123123 differs by .333, omitting those that are the same)

One possibility: 
At this point given that `(dtype().compatible(n.dtype())  && n.dtype().compatible(dtype()))` We know they are the same type. We can create logic for all bitwidth types like:
```
if (this->dtype().is_int8()
{
 int8_array a = this->value();
 int8_array b = n->value();
}
```

And then push the difference operation into the the DataArray class. 
a.diff(b,out);